datanode/sqlstore/positions.go

```diff
@@ -70,6 +71,27 @@ func (ps *Positions) GetByMarketAndParty(ctx context.Context,
 		marketID, partyID))
 }
 
+func (ps *Positions) GetByMarketAndParties(ctx context.Context, marketIDRaw string, partyIDsRaw []string) ([]entities.Position, error) {
+	marketID := entities.MarketID(marketIDRaw)
+	partyIDs := make([]entities.PartyID, 0, len(partyIDsRaw))
+	in := make([]stirng, 0, len(partyIDsRaw))
+	bindNum := 2
+	for _, p := range partyIDsRaw {
+		partyIDs = append(partyIDs, entities.PartyID(p))
+		in = append(in, fmt.Sprintf("$%d", bindNum))
+		bindNum++
+	}
+	bind := make([]interface{}, 0, len(in)+1)
+	// set all bind vars
+	bind = append(bind, marketID)
+	bind = append(bind, partyIDs...)
```

Comment:
We have a nice little helper called `nextBindVar` for this sort of thing

```go
	args := []interface{}{}

	conditions := []string{}
	if marketID.String() != "" {
		conditions = append(conditions, fmt.Sprintf("market_id=%s", nextBindVar(&args, marketID)))
	}
```

it sticks your argument onto an arg list and returns the right `$n` guy

Reply:
Yeah, I know about that helper, I just figured that having everything here, because I know the `len()` of the party ID's, this would be a bit more efficient, and because I'm using `IN()`, I just decided to do it all in-situ